src/wrappers/google/pubsub/SubscriptionWrapper.ts

```diff
@@ -0,0 +1,121 @@
+import ModuleUtils from '../../../utils/ModuleUtils';
+
+import Reporter from '../../../Reporter';
+import ConfigProvider from '../../../config/ConfigProvider';
+import ConfigNames from '../../../config/ConfigNames';
+import ExecutionContextManager from '../../../context/ExecutionContextManager';
+import ExecutionContext from '../../../context/ExecutionContext';
+
+import ThundraLogger from '../../../ThundraLogger';
+import { ClassNames, DomainNames } from '../../../Constants';
+import LambdaUtils from '../../../utils/LambdaUtils';
+
+import * as SubscriptionExecutor from './SubscriptionExecutor';
+import WrapperUtils from '../../WebWrapperUtils';
+
+const ApplicationClassName = ClassNames.NODEHANDLER;
+const ApplicationDomainName = DomainNames.API;
+
+let _REPORTER: Reporter;
+let _PLUGINS: any[];
+
+function subscriberListenWrapper(wrappedFunction: any) {
+    return function internalSubscriberListenWrapper() {
+        ThundraLogger.debug('<GoogleSubscriptionWrapper> Initializing ...');
+
+        const {
+            reporter,
+            plugins,
+        } = WrapperUtils.initWrapper(SubscriptionExecutor);
+
+        WrapperUtils.initAsyncContextManager();
+
+        _REPORTER = reporter;
+        _PLUGINS = plugins;
+
+        return wrappedFunction.apply(this);
+    };
+}
+
+function subscriberOnWrapper(wrappedFunction: any) {
+    return function internalPubSubSubscriberWrapper(eventName: any, callback: any) {
+        if (eventName !== 'message') {
+            return wrappedFunction.apply(this, [eventName, callback]);
+        }
+
+        const orginalCallback = callback;
+        const wrappedCallback = (message: any) => ExecutionContextManager.runWithContext(
+            () => {
+                return WrapperUtils.createExecContext(ApplicationClassName, ApplicationDomainName);
+            },
+            async function () {
+                if (!message) {
+                    return;
+                }
+
+                try {
+                    const context: ExecutionContext = this;
+                    ThundraLogger.debug('<GoogleSubscriptionWrapper> Before handling request');
+                    await WrapperUtils.beforeRequest(message, {}, _PLUGINS);
+
+                    await orginalCallback(message);
```

Comment:
It is better to have a `catch` block and in case of error, tag the span with the error and then throw the error.